### ISSUE TITLE
Running ant example causes Oracle javac to crash.

### ISSUE THREAD

**Issue by JakeWharton** [google/error-prone#287]
I'm experiencing problems with the compiler in my regular build, but in trying to craft a reproducible sample (by cannibalizing the example) I ran into the following exceptions internal to `javac` on both JDK 7 and 8.

Test script:
```bash
curl http://www.carfab.com/apachesoftware/ant/binaries/apache-ant-1.9.4-bin.zip > ant.zip
unzip ant.zip

TIMESTAMP=`curl https://oss.sonatype.org/content/repositories/snapshots/com/google/errorprone/error_prone_ant/2.0-SNAPSHOT/maven-metadata.xml | head -8 | tail -1 | sed  's/[^0-9.]//g'`
echo "Timestamp: ${TIMESTAMP}"
BUILD=`curl https://oss.sonatype.org/content/repositories/snapshots/com/google/errorprone/error_prone_ant/2.0-SNAPSHOT/maven-metadata.xml | head -9 | tail -1 | sed 's/[^0-9]//g'`
echo "Build number: ${BUILD}"

curl "https://oss.sonatype.org/content/repositories/snapshots/com/google/errorprone/error_prone_ant/2.0-SNAPSHOT/error_prone_ant-2.0-${TIMESTAMP}-${BUILD}.jar" > error_prone_ant.jar
mv error_prone_ant.jar apache-ant-1.9.4/lib/

git clone https://github.com/google/error-prone.git
cd error-prone/examples/ant

../../../apache-ant-1.9.4/bin/ant
```
If you create a temp directory somewhere you can quickly run this with:
```
curl https://gist.githubusercontent.com/JakeWharton/7bc348ceb1b9c27130f2/raw/676b2c18fbd95e46f6da9f85f4f474d459ecb956/test.sh | sh
```

Oracle JDK 8:
```
compile:
    [mkdir] Created dir: /Users/jw/Desktop/error-prone-test/error-prone/examples/ant/build
    [javac] Compiling 1 source file to /Users/jw/Desktop/error-prone-test/error-prone/examples/ant/build
    [javac] An exception has occurred in the compiler (1.8.0-internal_bootstrap). Please file a bug at the Java Developer Connection (http://java.sun.com/webapps/bugreport)  after checking the Bug Parade for duplicates. Include your program and the following diagnostic in your report.  Thank you.
    [javac] java.lang.NullPointerException
    [javac] 	at com.sun.tools.javac.jvm.CRTable.encodePosition(CRTable.java:168)
    [javac] 	at com.sun.tools.javac.jvm.CRTable.writeCRT(CRTable.java:115)
    [javac] 	at com.sun.tools.javac.jvm.ClassWriter.writeCode(ClassWriter.java:1170)
    [javac] 	at com.sun.tools.javac.jvm.ClassWriter.writeMethod(ClassWriter.java:1110)
    [javac] 	at com.sun.tools.javac.jvm.ClassWriter.writeMethods(ClassWriter.java:1602)
    [javac] 	at com.sun.tools.javac.jvm.ClassWriter.writeClassFile(ClassWriter.java:1692)
    [javac] 	at com.sun.tools.javac.jvm.ClassWriter.writeClass(ClassWriter.java:1620)
    [javac] 	at com.sun.tools.javac.main.JavaCompiler.genCode(JavaCompiler.java:746)
    [javac] 	at com.sun.tools.javac.main.JavaCompiler.generate(JavaCompiler.java:1572)
    [javac] 	at com.sun.tools.javac.main.JavaCompiler.generate(JavaCompiler.java:1536)
    [javac] 	at com.sun.tools.javac.main.JavaCompiler.compile2(JavaCompiler.java:901)
    [javac] 	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:860)
    [javac] 	at com.sun.tools.javac.main.Main.compile(Main.java:523)
    [javac] 	at com.sun.tools.javac.main.Main.compile(Main.java:381)
    [javac] 	at com.sun.tools.javac.main.Main.compile(Main.java:370)
    [javac] 	at com.google.errorprone.ErrorProneCompiler.run(ErrorProneCompiler.java:220)
    [javac] 	at com.google.errorprone.ErrorProneCompiler.run(ErrorProneCompiler.java:158)
    [javac] 	at com.google.errorprone.ErrorProneCompiler.compile(ErrorProneCompiler.java:87)
    [javac] 	at com.google.errorprone.ErrorProneAntCompilerAdapter.execute(ErrorProneAntCompilerAdapter.java:30)
    [javac] 	at org.apache.tools.ant.taskdefs.Javac.compile(Javac.java:1159)
    [javac] 	at org.apache.tools.ant.taskdefs.Javac.execute(Javac.java:935)
    [javac] 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
    [javac] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    [javac] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    [javac] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    [javac] 	at java.lang.reflect.Method.invoke(Method.java:483)
    [javac] 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
    [javac] 	at org.apache.tools.ant.Task.perform(Task.java:348)
    [javac] 	at org.apache.tools.ant.Target.execute(Target.java:435)
    [javac] 	at org.apache.tools.ant.Target.performTasks(Target.java:456)
    [javac] 	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
    [javac] 	at org.apache.tools.ant.Project.executeTarget(Project.java:1364)
    [javac] 	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
    [javac] 	at org.apache.tools.ant.Project.executeTargets(Project.java:1248)
    [javac] 	at org.apache.tools.ant.Main.runBuild(Main.java:851)
    [javac] 	at org.apache.tools.ant.Main.startAnt(Main.java:235)
    [javac] 	at org.apache.tools.ant.launch.Launcher.run(Launcher.java:280)
    [javac] 	at org.apache.tools.ant.launch.Launcher.main(Launcher.java:109)

BUILD FAILED
/Users/jw/Desktop/error-prone-test/error-prone/examples/ant/build.xml:21: Compile failed; see the compiler error output for details.
```

Oracle JDK 7:
```
compile:
    [javac] Compiling 1 source file to /Users/jw/Desktop/error-prone-test/error-prone/examples/ant/build
    [javac] An exception has occurred in the compiler (1.8.0-internal_bootstrap). Please file a bug at the Java Developer Connection (http://java.sun.com/webapps/bugreport)  after checking the Bug Parade for duplicates. Include your program and the following diagnostic in your report.  Thank you.
    [javac] java.lang.NoSuchFieldError: NATIVE_HEADER_OUTPUT
    [javac] 	at com.sun.tools.javac.file.Locations.initHandlers(Locations.java:654)
    [javac] 	at com.sun.tools.javac.file.Locations.<init>(Locations.java:92)
    [javac] 	at com.sun.tools.javac.util.BaseFileManager.createLocations(BaseFileManager.java:85)
    [javac] 	at com.sun.tools.javac.util.BaseFileManager.<init>(BaseFileManager.java:71)
    [javac] 	at com.sun.tools.javac.file.JavacFileManager.<init>(JavacFileManager.java:125)
    [javac] 	at com.sun.tools.javac.file.JavacFileManager$1.make(JavacFileManager.java:115)
    [javac] 	at com.sun.tools.javac.file.JavacFileManager$1.make(JavacFileManager.java:113)
    [javac] 	at com.sun.tools.javac.util.Context.get(Context.java:150)
    [javac] 	at com.sun.tools.javac.util.Context.get(Context.java:193)
    [javac] 	at com.sun.tools.javac.main.JavaCompiler.<init>(JavaCompiler.java:342)
    [javac] 	at com.google.errorprone.ErrorProneJavacJavaCompiler.<init>(ErrorProneJavacJavaCompiler.java:57)
    [javac] 	at com.google.errorprone.ErrorProneJavacJavaCompiler.<init>(ErrorProneJavacJavaCompiler.java:45)
    [javac] 	at com.google.errorprone.ErrorProneJavacJavaCompiler$1.make(ErrorProneJavacJavaCompiler.java:80)
    [javac] 	at com.google.errorprone.ErrorProneJavacJavaCompiler$1.make(ErrorProneJavacJavaCompiler.java:76)
    [javac] 	at com.sun.tools.javac.util.Context.get(Context.java:150)
    [javac] 	at com.sun.tools.javac.main.JavaCompiler.instance(JavaCompiler.java:86)
    [javac] 	at com.sun.tools.javac.main.Main.compile(Main.java:485)
    [javac] 	at com.sun.tools.javac.main.Main.compile(Main.java:381)
    [javac] 	at com.sun.tools.javac.main.Main.compile(Main.java:370)
    [javac] 	at com.google.errorprone.ErrorProneCompiler.run(ErrorProneCompiler.java:220)
    [javac] 	at com.google.errorprone.ErrorProneCompiler.run(ErrorProneCompiler.java:158)
    [javac] 	at com.google.errorprone.ErrorProneCompiler.compile(ErrorProneCompiler.java:87)
    [javac] 	at com.google.errorprone.ErrorProneAntCompilerAdapter.execute(ErrorProneAntCompilerAdapter.java:30)
    [javac] 	at org.apache.tools.ant.taskdefs.Javac.compile(Javac.java:1159)
    [javac] 	at org.apache.tools.ant.taskdefs.Javac.execute(Javac.java:935)
    [javac] 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
    [javac] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    [javac] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    [javac] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    [javac] 	at java.lang.reflect.Method.invoke(Method.java:606)
    [javac] 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
    [javac] 	at org.apache.tools.ant.Task.perform(Task.java:348)
    [javac] 	at org.apache.tools.ant.Target.execute(Target.java:435)
    [javac] 	at org.apache.tools.ant.Target.performTasks(Target.java:456)
    [javac] 	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
    [javac] 	at org.apache.tools.ant.Project.executeTarget(Project.java:1364)
    [javac] 	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
    [javac] 	at org.apache.tools.ant.Project.executeTargets(Project.java:1248)
    [javac] 	at org.apache.tools.ant.Main.runBuild(Main.java:851)
    [javac] 	at org.apache.tools.ant.Main.startAnt(Main.java:235)
    [javac] 	at org.apache.tools.ant.launch.Launcher.run(Launcher.java:280)
    [javac] 	at org.apache.tools.ant.launch.Launcher.main(Launcher.java:109)

BUILD FAILED
/Users/jw/Desktop/error-prone-test/error-prone/examples/ant/build.xml:21: Compile failed; see the compiler error output for details.
```

**Comment by JakeWharton** [comment]
Heh, carfab.com is apparently an apache mirror! Feel free to substitute any other (like http://apache.osuosl.org/ant/binaries/apache-ant-1.9.4-bin.zip).

**Comment by JakeWharton** [comment]
Specifically,
```
$ java -version
java version "1.8.0_25"
Java(TM) SE Runtime Environment (build 1.8.0_25-b17)
Java HotSpot(TM) 64-Bit Server VM (build 25.25-b02, mixed mode)
```
and
```
$ java -version
java version "1.7.0_72"
Java(TM) SE Runtime Environment (build 1.7.0_72-b14)
Java HotSpot(TM) 64-Bit Server VM (build 24.72-b04, mixed mode)
```

**Comment by cushon** [comment]
Nice repro! I have a fix for the JDK8 part: #288

**Comment by JakeWharton** [comment]
:+1: Yep that fixed 8.

**Comment by cushon** [comment]
I'm not sure how to fix JDK7 yet. Error-prone now repackages javac 8, which only works on JDK7 if it can override certain classes in `rt.jar`. That requires either setting `-J-Xbootclasspath/p:somepath/javac.jar` or doing hacky classloading. There's probably a way to configure ant to do that, but JDK8 is going to be the happy path until it gets sorted out.

Sorry for the problems you're encountering. Our build system integrations have been neglected. I'm fixing maven now, and ant was next on the list.

**Comment by JakeWharton** [comment]
All our stuff that I'm using this with is on 8, so it's not a huge deal. Switching to 7 was just a thing I figured I'd try to see whether it was a version-specific problem (I even tried 6 which got absolutely nowhere). As far as I'm concerned this issue is resolved.

Also not at all an Ant user, I just figured it would be the easiest example to adapt. We're mostly Maven with some Gradle for our Android client apps. I'm currently working in the Gradle integration with the latest 2.0 stuff building off of @tbroyer's plugin.
